### DISCUSSION TITLE
Compiler error

### DISCUSSION THREAD

**Discussion by strasb** [cyberman54/ESP32-Paxcounter#880] [Q&A] [answered]
Hello.

I new to everything when it comes to compiling on anything so im a bit lost on this error message im reciveing. All software is up to date btw. 

src/i2c.cpp: In function 'void i2c_scan()':
src/i2c.cpp:75:48: error: too few arguments to function 'int I2CDiscoverDevice(BBI2C*, uint8_t, uint32_t*)'
           iDevice = I2CDiscoverDevice(&bbi2c, i);
                                                ^
In file included from include/i2c.h:5:0,
                 from src/i2c.cpp:3:
.pio/libdeps/usb/BitBang_I2C/src/BitBang_I2C.h:151:5: note: declared here
 int I2CDiscoverDevice(BBI2C *pI2C, uint8_t i, uint32_t *pCapabilities);
     ^
*** [.pio\build\usb\src\i2c.cpp.o] Error 1
==================== [FAILED] Took 34.06 seconds ====================

Environment    Status    Duration
-------------  --------  ------------
usb            FAILED    00:00:34.062
=============== 1 failed, 0 succeeded in 00:00:34.062 =============== 

I have 5 errors  iin my i2c.ccp file it seems. I've have not changed anything in this file.

--

**Comment by strasb** [comment]
Cloned the repository instead of copying it manually and now it works, have no idea why =D

**Comment by cyberman54** [comment, marked as answer]
@strasb Looks like you had an outdated version of code on your disc.
The error in i2c.cpp comes from updating onebitdisplay lib to v2.x, this needed a change in line 75 of i2c.cpp.
This was made with a [commit](https://github.com/cyberman54/ESP32-Paxcounter/commit/30b56c8bcffd654885735ecf97f392369ae09308) an 3 Jun 2022. 

Thus, update your repository.

If you made a local copy, either by just downloading the code, or by cloning the repo, you must **pull** the current code.
